test/metabase/db/custom_migrations_test.clj

```diff
@@ -2177,3 +2177,168 @@
                      (json/parse-string keyword-except-column-key)))))))))
+
+(deftest set-stage-number-in-parameter-mappings-test
+  (testing "v52.2024-10-26T18:42:42"
+    (impl/test-migrations ["v52.2024-10-26T18:42:42"] [migrate!]
+      (let [single-stage-query {:source-table 29, :filter [:> [:field 288 nil] "2015-01-01"]},
+            multi-stage-query {:source-query
+                               {:source-query single-stage-query
+                                :aggregation [:count],
+                                :order-by [[:asc [:field 275 {:source-field 290}]]],
+                                :breakout [[:field 200 nil]
+                                           [:field 275 {:source-field 290}]],
+                                :filter [:starts-with [:field 275 {:source-field 290}] "F"]},
+                               :filter [:> [:field "count" {:base-type :type/Integer}] 5]}
+            native-query {:query "SELECT ID, NAME, CATEGORY_ID FROM VENUES ORDER BY ID DESC LIMIT 2"}
+            user-id (t2/insert-returning-pks! (t2/table-name :model/User)
+                                              {:first_name  "Howard"
+                                               :last_name   "Hughes"
+                                               :email       "howard@aircraft.com"
+                                               :password    "superstrong"
+                                               :date_joined :%now})
+            database-id (t2/insert-returning-pks! (t2/table-name :model/Database)
+                                                  {:name       "DB"
+                                                   :engine     "h2"
+                                                   :created_at :%now
+                                                   :updated_at :%now
+                                                   :details    "{}"})
+            dashboard-id (t2/insert-returning-pks! :model/Dashboard {:name                "My Dashboard"
+                                                                     :creator_id          user-id
+                                                                     :parameters          []})
+            single-stage-dataset-query (json/generate-string {:type "query"
+                                                              :database database-id
+                                                              :query single-stage-query})
+            multi-stage-dataset-query (json/generate-string {:type "query"
+                                                             :database database-id
+                                                             :query multi-stage-query})
+            native-dataset-query (json/generate-string {:type "native"
+                                                        :database database-id
+                                                        :native native-query})
+            parameter-mappings-without-stage-numbers [{:card_id 1
```

Comment:
In practice would `:card_id` be consistent with the card its dashcard is linked to? Wondering it isn't worth making this a function so that the data is all consistent.

Reply:
Opened #49429 to address this.